datasets/doc/source/how-to-use-with-local-data.rst

```diff
@@ -0,0 +1,226 @@
+Use with Local Data
+===================
+
+You can partition data from local files and in-memory data in
+``Flower Datasets`` library using ``Partitioner`` s.
+This guide details how to load and partition your dataset created from:
+
+* local files: CSV, JSON, image, audio data,
+* in-memory data : dictionary, list, pd.DataFrame, np.ndarray.
+
+
+General Overview
+----------------
+FederatedDataset is an abstraction that performs all the steps to prepare the dataset for FL experiments: downloading, preprocessing (including resplitting), and partitioning.
+However, the partitioning happens thanks to the ``partitioners`` of type ``str: Partitioner`` (meaning a dictionary mapping the split name of the dataset to the ``Partitioner``).
```

Comment:
I'm unsure if we need this level of detail for this guide.

Reply:
Ok, I simplified the General Overview section.